### PR TITLE
If a controller is specified for a component class using an Expander, don't normalize or "request" pins

### PR DESCRIPTION
**Pull request by rwaldron**
cc @dtex 

This will pull into `controllers-can-use-expanders`

### CODE REVIEW

**Comment by scottgonzalez** [comment]
Can `Board.Component` just do that check internally and not require each component to repeat the logic?

**Comment by rwaldron** [comment]
> Can Board.Component just do that check internally and not require each component to repeat the logic?

Well, yes... I wanted it that way to begin with and wrote it as such, but... putting this in `board.js` requires adding: 

```js
var Expander = require("../lib/expander");
```

And inside `expander.js` there is: 

```js
var Board = require("../lib/board");
```

And some time later...

```js
var methods = Object.keys(Board.prototype);
```

Which causes: 

```
TypeError: Cannot convert undefined or null to object
```

Let me try to get around that.

File: lib/board.js [hunk 1]
```diff
@@ -818,6 +819,17 @@ Board.Component = function(opts, componentOpts) {
     }
   }
 
+  if (!Expander) {
```
Comment (negherbon) [inline lib/board.js]: with instruction inside, is better remove keys or is the same thing?

**Comment by dtex** [comment]
When initializing a component on an expander we still need to run the expander's pin normalization on the component's opts.pin/opts.pins. I don't see that here but maybe I'm just missing it?

**Comment by dtex** [comment]
Did some testing. Looks pretty good but doesn't warn about pin reuse on controllers. This should warn but does not:

````
  var servo2 = new five.Servo({
    controller: "PCA9685",
    address: 0x73,
    pin: 2
  });

  var servo3 = new five.Servo({
    controller: "PCA9685",
    address: 0x73,
    pin: 2
  });
````

**Comment by rwaldron** [comment]
>  but doesn't warn about pin reuse on controllers. This should warn but does not:

I don't know what to do about this right now. TBH, pin tracking has been nothing but a nightmare.

**Pull request merged**